### DISCUSSION TITLE
Single mode oneshot with one song

### DISCUSSION THREAD

**Discussion by SoongNoonien** [MusicPlayerDaemon/MPD#1855] [Q&A] [answered]
Hi!
If the queue contains exactly one song and single mode oneshot is active it will remain active even after the song has finished. @jcorporation I reckon you implemented this feature so maybe you know if this behaviour is intended? I would expect single mode to be off after the song has finished. It seems like only the automatic beginning of a next song triggers the deactivation of the single mode oneshot and a stop/pause command.

Martin

**Comment by jcorporation** [comment, marked as answer]
Yes, this is intended. Consume removes the song in the playlist::QueuedSongStarted function. After the song was removed we disable the oneshot mode.

https://github.com/MusicPlayerDaemon/MPD/blob/master/src/queue/Playlist.cxx

**Reply by SoongNoonien** [reply]
Ok, thanks for the quick answer.
